mobile/src/main/java/rs/readahead/washington/mobile/views/fragment/resources/ResourcesListFragment.kt

```diff
@@ -0,0 +1,354 @@
+                    null
+                )
+            )
+
+            if (!isDownloaded) {
+                dlOpenButton.setImageDrawable(
+                    ResourcesCompat.getDrawable(
+                        resources,
+                        R.drawable.ic_download,
+                        null
+                    )
+                )
+
+                dlOpenButton.contentDescription = getString(R.string.action_download)
+
+                dlOpenButton.setOnClickListener {
+                    if (MyApplication.isConnectedToInternet(requireContext())) {
+                        dlOpenButton.hide()
+                        downloading.show()
+                        model.downloadResource(resource)
+                    } else {
+                        DialogUtils.showBottomMessage(
+                            baseActivity,
+                            getString(R.string.collect_blank_toast_not_connected),
+                            true
+                        )
+                    }
+                }
+            } else {
+                dlOpenButton.setImageDrawable(
+                    ResourcesCompat.getDrawable(
+                        resources,
+                        R.drawable.ic_more,
+                        null
+                    )
+                )
+
+                dlOpenButton.contentDescription =
+                    getString(R.string.collect_blank_action_desc_more_options)
+
+                dlOpenButton.setOnClickListener {
+                    showDownloadedMenu(
+                        resource
+                    )
+                }
+
+                row.setOnClickListener {
+                    model.getPdfFile(resource.fileId)
+                }
+            }
+        }
+        return itemBinding.root
+    }
+
+    private fun showDownloadedMenu(resource: Resource) {
+        BottomSheetUtils.showViewDeleteMenuSheet(
+            requireActivity().supportFragmentManager,
+            resource.title,
+            requireContext().getString(R.string.View_Report),
+            requireContext().getString(R.string.Resources_RemoveFromDownloads),
+            object : BottomSheetUtils.ActionSeleceted {
+                override fun accept(action: BottomSheetUtils.Action) {
+                    if (action === BottomSheetUtils.Action.VIEW) {
+                        model.getPdfFile(resource.fileId)
+                    }
+                    if (action === BottomSheetUtils.Action.DELETE) {
+                        model.removeResource(resource)
+                        deletedResource(resource)
+                    }
+                }
+            },
+            requireContext().getString(R.string.Resources_RemoveFromDownloads),
+            String.format(
+                requireContext().resources.getString(R.string.Collect_Subtitle_RemoveForm),
+                resource.title
+            ),
+            requireContext().getString(R.string.action_remove),
```

Comment:
Consider using injected  `baseActivity` instead of `requireContext()` this can lead to potential crashs.